qiskit_machine_learning/datasets/gaussian.py

```diff
@@ -22,138 +22,102 @@
-
-        training_feature_array, training_label_array = features_and_labels_transform(
-            training_input, class_labels, one_hot)
-        test_feature_array, test_label_array = features_and_labels_transform(
-            test_input, class_labels, one_hot)
-
-        if plot_data:
-            try:
-                import matplotlib.pyplot as plt
-            except ImportError as ex:
-                raise MissingOptionalLibraryError(
-                    libname='Matplotlib',
-                    name='gaussian',
-                    pip_install='pip install matplotlib') from ex
-
-            for k in range(0, 3):
-                plt.scatter(sample_train[label_train == k, 0][:training_size],
-                            sample_train[label_train == k, 1][:training_size])
-
-            plt.title("Gaussians")
-            plt.show()
-
-        return training_feature_array, training_label_array, test_feature_array, test_label_array
-    else:
+    if n not in (2, 3):
         raise ValueError("Gaussian presently only supports 2 or 3 qubits")
+
+    class_labels = _generate_class_labels(n)
+    label_train, randomized_vectors, samples_train = _init_data_structures(n, test_size,
+                                                                           training_size)
+    sigma = 1
+
+    for t_r in range(training_size + test_size):
+        for feat in range(n):
+            for sample, randomized_vector in zip(samples_train, randomized_vectors):
+                sample[t_r][feat] = _calc_random_normal(randomized_vector, feat,
+                                                        sigma) if n == 2 \
+                    else _calc_random_normal_2(randomized_vector, feat, sigma)
+                # TODO unify both functions in the future when other n's supported
+
+    _update_label_train(n, label_train, test_size, training_size)
+    label_train = label_train.astype(int)
+
+    samples_train = np.reshape(samples_train, (n * (training_size + test_size), n))
+    training_input = {key: (samples_train[label_train == k, :])[:training_size]
+                      for k, key in enumerate(class_labels)}
+    test_input = {
+        key: (samples_train[label_train == k, :])[training_size:(training_size + test_size)] for
+        k, key in enumerate(class_labels)}
+
+    training_feature_array, training_label_array = features_and_labels_transform(
+        training_input, class_labels, one_hot)
+    test_feature_array, test_label_array = features_and_labels_transform(
+        test_input, class_labels, one_hot)
+
+    if plot_data:
+        _plot(n, label_train, samples_train, training_size)
+
+    return training_feature_array, training_label_array, test_feature_array, test_label_array
+
+
+def _generate_class_labels(n):
+    if n > 25:
+        # TODO change to it a more complex labeling system e.g. AA, AB, AAA etc. or just to C1,
+        #  C2, C3 etc.
+        raise Exception("To many classes requested. Maximum is 26")
```

Comment:
Should raise something like ValueError not a raw Exception. Also its informative to have the failing value included in the message.

Reply:
I agree, will fix that.

Reply:
FYI There was another exception in there that could have the value in the message too - I just did not call it out explicitly,